### PR TITLE
More ^C handling in RCons, but still not working with agf because of the nestness nature of it

### PR DESCRIPTION
**Pull request by radare**

### CODE REVIEW

File: libr/core/agraph.c [hunk 1]
```diff
@@ -3083,6 +3089,7 @@ static void agraph_print_edges(RAGraph *g) {
 		}
 	}
 
+	// XXX why not use r_list_free ??
```
Comment (lowlyw) [inline libr/core/agraph.c]: why just comment, why not do it? ;)

File: libr/cons/canvas_line.c [hunk 2]
```diff
@@ -434,6 +434,9 @@ R_API void r_cons_canvas_line_back_edge (RConsCanvas *c, int x, int y, int x2, i
 		r_cons_canvas_line (c, x, y, x2, y2, style);
 		return;
 	}
+	if (r_cons_is_breaked ()) {
```
Comment (lowlyw) [inline libr/cons/canvas_line.c]: this isn't in a loop. why do we need it here?

**Comment by radare** [comment]
@lowlyw the problem is that each consbreak/push/pop is breaking at one level only, if the caller checks again for that it will fail to interrupt all the task, only a slow loop inside. so ^C is not fixed in agf. Do you want to finish this PR?

**Comment by radare** [comment]
closing for now, preformance is better now and i dont see he need to bloat those loops

**Pull request closed without merging**